pennylane/operation.py

```diff
@@ -285,24 +286,42 @@ def __init__(self, *args, wires=None, do_queue=True):
             assert self.grad_recipe is None, 'Gradient recipe is only used by the A method!'
 
         # apply the operation on the given wires
-        if isinstance(wires, int):
-            wires = [wires]
-
-        if self.num_wires != 0 and len(wires) != self.num_wires:
-            raise ValueError("{}: wrong number of wires. "
-                             "{} wires given, {} expected.".format(self.name, len(wires), self.num_wires))
+        if not isinstance(wires, Sequence):
+            self._wires = [wires]
+        else:
+            self._wires = wires
```

Comment:
The new `self._wires` attribute stores the wires as a sequence of both `ints` and `Variables`. To dynamically determine the list of wires as integers, the new `self.wires` property (replacing the old `self.wires` attribute, but returning the same result for compatibility) must be accessed